### PR TITLE
upgrade delta-sharing-spark to scala 2.13

### PR DESCRIPTION
**Pull request by linzhou-db**
upgrade delta-sharing-spark to scala 2.13

### CODE REVIEW

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala [hunk 1]
```diff
@@ -143,7 +143,7 @@ case class DeltaSharingSource(
 
   private var lastGetVersionTimestamp: Long = -1
   private var lastQueriedTableVersion: Long = -1
-  private val QUERY_TABLE_VERSION_INTERVAL_MILLIS = 30000 // 30 seconds
+  private val QUERY_TABLE_VERSION_INTERVAL_MILLIS = 3000 // 30 seconds
```
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala:146]: Is this intentional?

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingFileProfileProviderSuite.scala [hunk 2]
```diff
@@ -70,8 +70,8 @@ class DeltaSharingFileProfileProviderSuite extends SparkFunSuite {
     )
   }
 
-  test("version is missing") {
-    val e = intercept[IllegalArgumentException] {
+  test("shareCredentialsVersion is missing") {
+    val e = intercept[Exception] {
```
Comment (chakankardb) [inline spark/src/test/scala/io/delta/sharing/spark/DeltaSharingFileProfileProviderSuite.scala:74]: Did the type of exception change?

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceCDFSuite.scala [hunk 3]
```diff
@@ -563,21 +563,26 @@ class DeltaSharingSourceCDFSuite extends QueryTest
...
+        .start()
       // There are 5 checkpoints for 5 getBatch(), which is caused by maxFilesPerTrigger = 1,
       // remove the latest 3 checkpoints.
       val checkpointFiles = FileUtils.listFiles(checkpointDir, null, true).asScala
       checkpointFiles.foreach{ f =>
         if (!f.isDirectory() && (f.getCanonicalPath.endsWith("2") ||
-          f.getCanonicalPath.endsWith("3") || f.getCanonicalPath.endsWith("4"))) {
+          f.getCanonicalPath.endsWith("3") || f.getCanonicalPath.endsWith("4") ||
+          f.getCanonicalPath.endsWith("2.crc") || f.getCanonicalPath.endsWith("3.crc") ||
+          f.getCanonicalPath.endsWith("4.crc") || f.getCanonicalPath.endsWith("metadata.crc"))) {
```
Comment (chakankardb) [inline spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceCDFSuite.scala:578]: Question: why were these needed for a scala version change?

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala [hunk 4]
```diff
@@ -446,29 +446,35 @@ class DeltaSharingSourceSuite extends QueryTest
...
         if (!f.isDirectory() &&
-          (f.getCanonicalPath.endsWith("2") || f.getCanonicalPath.endsWith("3"))) {
+          (f.getCanonicalPath.endsWith("2") || f.getCanonicalPath.endsWith("3") ||
+            f.getCanonicalPath.endsWith("2.crc") || f.getCanonicalPath.endsWith("3.crc") ||
+            f.getCanonicalPath.endsWith("metadata.crc"))) {
           f.delete()
         }
+        if (f.getCanonicalPath.endsWith("metadata")) {
+          FileUtils.writeStringToFile(f, s"""{"id":"${newQuery.id}"}""")
+        }
```
Comment (chakankardb) [inline spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala:466]: These seem unrelated to scala version change?

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala (hunk 1 above)
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala]: good catch, I'll revert.

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingFileProfileProviderSuite.scala (hunk 2 above)
Comment (linzhou-db) [reply spark/src/test/scala/io/delta/sharing/spark/DeltaSharingFileProfileProviderSuite.scala:74]: Yes, it changes with the new scala version.

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceCDFSuite.scala (hunk 3 above)
Comment (linzhou-db) [reply spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceCDFSuite.scala:578]: Because with the new scala version, the spark structured streaming requires the query id to match to reuse a checkpoint. 

Here I want to test the checkpoint, I'm not able to produce new data, so I have to delete some files from the checkpoint to manually roll back the process. Thus the deletion of these new files are needed for the new scala version.

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala (hunk 4 above)
Comment (linzhou-db) [reply spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala:467]: It's related as explained above, SparkStructuredStreaming requires query id match to reuse checkpoint.

**Review by chakankardb — APPROVED** [review]
@zsxwing Can you please take a close look as well?

**Comment by linzhou-db** [comment]
moved to https://github.com/delta-io/delta-sharing/pull/333
